examples/quantum_die.py

```diff
@@ -28,47 +28,43 @@ def qubits_needed(n):
     """
     return int(math.ceil(math.log(n, 2)))
 
+def qvm(n):
+    """
+    Connect to a QVM that simulates n sides.
+    """
+    if qubits_needed(n) == 9:
+        return qvm = get_qc('9q-square-qvm')
+    else:
+        return qvm = get_qc('%(qubits)dq-qvm' % {"qubits": qubits_needed(n)})
+
 def die_program(n):
     """
     Generate a quantum program to roll a die of n faces.
     """
-    prog = pq.Program()
+    prog = Program()
     qubits = qubits_needed(n)
+    ro = prog.declare('ro', 'BIT', qubits)
     # Hadamard initialize.
     for q in range(qubits):
         prog.inst(H(q))
     # Measure everything.
     for q in range(qubits):
-        prog.measure(q, [q])
-    return prog
+        prog.measure(q, ro[q])
+    return qvm(n).compile(prog)
 
-def process_result(r):
+    
+def roll_die(n):
     """
-    Convert a list of measurements to a die value.
+    Roll an n-sided quantum die.
     """
-    return reduce(lambda s, x: 2*s + x, r, 0)
-
-BATCH_SIZE = 10
+    results = qvm(n).run(die_program(n))
+    return results
 
-dice = {}
-
-qvm = api.QVMConnection()
-
-def roll_die(n):
+def process_result(results):
     """
-    Roll an n-sided quantum die.
+    Convert a list of measurements to a die value.
     """
-    addresses = list(range(qubits_needed(n)))
-    if not n in dice:
-        dice[n] = die_program(n)
-    die = dice[n]
-    # Generate results and do rejection sampling.
-    while True:
-        results = qvm.run(die, addresses, BATCH_SIZE)
-        for r in results:
-            x = process_result(r) + 1
-            if 0 < x <= n:
-                return x
+    return reduce(lambda x, y: 2*x + y, results[0], 0) + 1
```

Comment:
This is pretty slick! Since this is an example file that newbies might like to look at, it might be a good idea to at least add a comment explaining how this line does what we want, or to split it up into multiple lines.

Reply:
oldbies (me) may also find this hard to read

Reply:
(tbh same)